### PR TITLE
Adding the CLI package

### PR DESCRIPTION
**Pull request by ramitha**
This takes care of all the comments received in discarded pull request.

### CODE REVIEW

File: packages/@icon-magic/cli/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+- **iconrc.(js|json)** A config file for a single icon or a group of icons with
+  paths to the various icon directories and their variants at minimum. The
+  config further be caustomaized
+- **Flavor**: An asset obtained after applying build/generate plugins on the
+  source .svg file. A flavor will also contain assets for each type, i. e.,
+  paths to the .svg, .png and .webp version of that flavor.
+
+## Usage
+
+For a detailed list of the terminology, refer
```
Comment (oa495) [inline packages/@icon-magic/cli/README.md]: refer to the glossary?

File: packages/@icon-magic/cli/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    if (!inputPaths.length) {
+      LOGGER.error(
+        "No Input Directories were specified.\nDid you mean 'icon-magic generate .'?"
+      );
+    }
+    // Get the iconSet from the inputPaths
+    const iconSet = configReader.getIconConfigSet(inputPaths);
+
+    // generate all the icons
+    await iconGenerate.generateFromConfigHash(iconSet);
```
Comment (oa495) [inline packages/@icon-magic/cli/src/index.ts]: is this supposed to be `generate`? No `generateFromConfigHash` currently exists in `generate`

Comment (ramitha) [reply packages/@icon-magic/cli/src/index.ts]: Now it does :) I just pushed https://github.com/linkedinlabs/icon-magic/pull/18

File: packages/@icon-magic/cli/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  build               Construct flavors that an icon from its variants, after applying the build plugins.
+  generate            Generates the flavors of the icon in the extension types that it can be consumed.
+  distribute          Moves an icon from the source to the destination, applying plugins if specified.
+
+Options:
+  -c, --config        Path to config file name
+  -i, --inputPath     Path to the input directory of icons
+  -o, --outputPath    Path to the output directory where the generated assets are to be written to
+  -h, --help          Display usage
+  -v, --version       Display version
```
Comment (zimmi88) [inline packages/@icon-magic/cli/README.md]: Minor nitpick: You'll need to add `program.version('1.2.3');` in order for commander to output version.

https://www.npmjs.com/package/commander#version-option

**Pull request merged**